File: src/main/java/kitchenpos/products/tobe/domain/ProductRepository.java
package kitchenpos.products.tobe.domain;


import java.util.List;
import java.util.Optional;

public interface ProductRepository {
    Product save(Product product);

    Optional<Product> findById(Long id);

    List<Product> findAll();

    List<Product> findAllByIdIn(List<Long> ids);
}

File: src/main/java/kitchenpos/products/tobe/domain/ProductRepository.kt
package kitchenpos.products.tobe.domain

import java.util.*

interface ProductRepository {
    fun save(product: Product): Product

    fun findById(id: UUID): Product?

    fun findAll(): List<Product>

    fun findAllByIdIn(ids: List<UUID>): List<Product>

    fun flush()
}
